src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -0,0 +1,268 @@
+    private final JButton searchModeButton = new JButton();
+    private final JToggleButton caseSensitive;
+    private final JToggleButton regularExp;
+
+    private final SearchQueryHighlightObservable searchQueryHighlightObservable = new SearchQueryHighlightObservable();
+
+    private SearchWorker searchWorker;
+    private AutoCompleteSupport<String> autoCompleteSupport = new AutoCompleteSupport<>(searchField);
+
+    private final SearchPreferences searchPreferences;
+    private SearchDisplayMode searchDisplayMode;
+
+
+    public GlobalSearchBar() {
+        super();
+        searchPreferences = new SearchPreferences(Globals.prefs);
+        searchDisplayMode = searchPreferences.getSearchMode();
+
+        currentResults.setFont(currentResults.getFont().deriveFont(Font.BOLD));
+        searchField.setColumns(30);
+
+        regularExp = new JToggleButton(IconTheme.JabRefIcon.REG_EX.getSmallIcon(),
+                searchPreferences.isRegularExpression());
+        regularExp.setToolTipText(Localization.lang("regular expression"));
+        regularExp.addActionListener(e -> {
+            searchPreferences.setRegularExpression(regularExp.isSelected());
+            performSearch();
+        });
+
+        caseSensitive = new JToggleButton(IconTheme.JabRefIcon.CASE_SENSITIVE.getSmallIcon(),
+                searchPreferences.isCaseSensitive());
+        caseSensitive.setToolTipText(Localization.lang("Case sensitive"));
+        caseSensitive.addActionListener(e -> {
+            searchPreferences.setCaseSensitive(caseSensitive.isSelected());
+            performSearch();
+        });
+
+        updateSearchModeButtonText();
+        searchModeButton.addActionListener(l -> {
+            int nextSearchMode = (searchDisplayMode.ordinal() + 1) % SearchDisplayMode.values().length;
+            searchDisplayMode = SearchDisplayMode.values()[nextSearchMode];
+            searchPreferences.setSearchMode(searchDisplayMode);
+            updateSearchModeButtonText();
+            performSearch();
+        });
+
+        clearSearchButton.setToolTipText(Localization.lang("Clear"));
+        clearSearchButton.addActionListener(l -> endSearch());
+
+        searchField.addFocusListener(Globals.focusListener);
+        searchField.addActionListener(e -> performSearch());
+        JTextFieldChangeListenerUtil.addChangeListener(searchField, e -> performSearch());
+
+        String endSearch = "endSearch";
+        searchField.getInputMap().put(Globals.getKeyPrefs().getKey(KeyBinding.CLEAR_SEARCH), endSearch);
+        searchField.getActionMap().put(endSearch, new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                System.out.println(endSearch);
+                endSearch();
+            }
+        });
+
+        autoCompleteSupport.install();
+
+        setLayout(new WrapLayout(FlowLayout.RIGHT));
+        toolBar.setFloatable(false);
+        toolBar.add(currentResults);
```

Comment:
Is it really necessary to display the found results in front of the search field? It takes up a lot of space in the toolbar and when having a small resoultion or small screen it might not even be visible as shown below.
![screen shot 2016-07-05 at 14 28 16](https://cloud.githubusercontent.com/assets/15340757/16584748/cec164a0-42bd-11e6-9680-959aa41d1e66.png)

So why not update the status text at the very bottom of the main frame?